cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -109,6 +114,96 @@ cros_gralloc_driver *cros_gralloc_driver::get_instance()
         close(fd);                   \
     }
 
+int32_t cros_gralloc_driver::reload()
+{
+	int fd;
+	drmVersionPtr version;
+	char const *str = "%s/renderD%d";
+	uint32_t min_node = 128;
+	uint32_t max_node = (min_node + drv_num_ + IVSH_DEVICE_NUM);
+	uint32_t j;
+	char *node;
+
+	const int render_num = 10;
+	const int name_length = 50;
+	int availabe_node = 0;
+	uint32_t gpu_grp_type = 0;
+
+	availabe_node = 0;
+	for (uint32_t i = min_node + drv_num_; i < max_node; i++) {
+		if (asprintf(&node, str, DRM_DIR_NAME, i) < 0)
+			continue;
+
+		fd = open(node, O_RDWR, 0);
+		free(node);
+		if (fd < 0)
+			continue;
+
+		version = drmGetVersion(fd);
+		if (!version) {
+			close(fd);
+			continue;
+		}
+
+		availabe_node++;
+		close(fd);
+		drmFreeVersion(version);
+	}
+
+	// have the ivshemem node
+	if ((availabe_node > 0) && !drv_ivshmem_) {
+		drv_num_ = drv_num_ + availabe_node;
+		if (asprintf(&node, str, DRM_DIR_NAME, (min_node + drv_num_ - 1)) < 0) {
+			drv_loge("Open new added node fail\n");
+			return -ENODEV;
+		}
+
+		fd = open(node, O_RDWR, 0);
+		free(node);
+
+		if (fd < 0) {
+			return -ENODEV;
+		}
+
+		version = drmGetVersion(fd);
+		if (!version) {
+			close(fd);
+			return -ENODEV;
+		}
+
+		if (strcmp(version->name, "virtio_gpu")) {
+			drv_logi("New added node is not virtiogpu device\n");
+			close(fd);
+			drmFreeVersion(version);
+			return -ENODEV;
+		}
+
+		drv_ivshmem_ = drv_create(fd);
+		if (!drv_ivshmem_) {
+			drv_loge("Failed to create driver for virtio device\n");
+			close(fd);
+			return -ENODEV;
+		}
+		gpu_grp_type = GPU_WITH_IVSHMEM;
+
+		if (drv_ivshmem_) {
```

Comment:
Looks like this check is redundant since it was already checked in line 182 already.